tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -129,12 +137,33 @@ def tags_route(self, request):
     @wrappers.Request.application
     def scalars_route(self, request):
         """Given a tag and single run, return array of ScalarEvents."""
-        tag = request.args.get("tag")
-        run = request.args.get("run")
+        if request.method == "GET":
```

Comment:
I drafted #3838 on Friday but wasn’t sure whether to send it; after
seeing the increased scope of the HTTP-specific parts of the request
handling in this PR, I think that it would be useful to land that first
so that this PR can be better tested. Does that sound reasonable to you?